services/net/clip/ClipAction.cs

```diff
@@ -398,9 +415,34 @@ private async Task<long> GetDurationAsync(string inputFile)
         var output = await process.StandardOutput.ReadToEndAsync();
         await process.WaitForExitAsync();
 
-        var value = String.IsNullOrWhiteSpace(output) ? 0 : float.Parse(output);
+        if (String.IsNullOrWhiteSpace(output)) return 0;
+        return float.TryParse(output, out float value) ? (long)Math.Floor(value) : 0;
+    }
+
+    /// <summary>
+    /// Repackage the file into a different type.
+    /// </summary>
+    /// <param name="inputFile"></param>
+    /// <param name="format"></param>
+    /// <returns></returns>
+    private static async Task<string> RepackageFileAsync(string inputFile, string format = "mp4")
```

Comment:
Need to convert `mkv` into `mp4` to get the duration of the video regrettably.  There are warnings thrown and an error during this process that need to be reviewed, but it appears to function correctly.